ast.go

```diff
@@ -3813,7 +3813,12 @@ type ParenExpr struct {
 }
 
 // String returns a string representation of the parenthesized expression.
-func (e *ParenExpr) String() string { return fmt.Sprintf("(%s)", e.Expr.String()) }
+func (e *ParenExpr) String() string {
+	if e.Expr == nil || e == nil {
```

Comment:
These have to be reversed: nil check for `e` first, then for the field of `e`

Reply:
Otherwise `e.Expr` will panic on a nil `e`

Reply:
Sounds good--I'll go through and add a check to the various `String()` methods that are missing them, it appears we have nil checks in a few of them.